### PR TITLE
Remove Unicode dependency to avoid large downloads for mathlib4 users

### PR DESCRIPTION
**Pull request by eric-wieser**
Until `lake` learns to not download this, I would claim that the benefit of marginally-nicer URLs is not worth the cost of having every mathlib4 user download a large archive of unicode data.

### CODE REVIEW

File: DocGen4/Output/DocString.lean [hunk 1]
```diff
@@ -55,9 +54,7 @@ partial def xmlGetHeadingId (el : Xml.Element) : String :=
       |>.filter (!·.isEmpty)
       |> replacement.intercalate
     unicodeToDrop (c : Char) : Bool :=
-      charInGeneralCategory c GeneralCategory.punctuation ||
-      charInGeneralCategory c GeneralCategory.separator ||
-      charInGeneralCategory c GeneralCategory.other
+      false -- TODO: restore the behavior described in the docstring
```
Comment (eric-wieser) [inline DocGen4/Output/DocString.lean:57]: `c == " "` is probably a marginally better choice

Comment (hargoniX) [reply DocGen4/Output/DocString.lean:57]: Sure, go for it then :+1:

**Comment by hargoniX** [comment]
Do you want to make the mentioned change in the comment or shall I go ahead and merge it?

**Comment by eric-wieser** [comment]
CI fails so I would recommend against merging this!

**Comment by hargoniX** [comment]
Oh I was thinking this is just the usual CI fail from mathlib4 and std4 incompatability /o\

**Comment by fgdorais** [comment]
Isn't this mostly fixed by https://github.com/leanprover/doc-gen4/commit/5f59fbaac4b2a070d57d642f42f812641b6f8163

**Comment by hargoniX** [comment]
It's definitely a great improval yes, I'll close the PR for now.

**Pull request closed without merging**